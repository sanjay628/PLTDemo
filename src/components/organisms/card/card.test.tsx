import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CardView from './card.component';
const createTestProps = (props: Object) => ({
  ...props,
});
describe('CardView', () => {
  let props: any;
  beforeEach(() => {
    props = createTestProps({});
  });
  it('renders item name, price, and Stepper component on cart screen', () => {
    const onIncQtyMock = jest.fn();
    const onDecQtyMock = jest.fn();
    const onCartMock = jest.fn();
    const {getByText} = render(
      <CardView
        itemName="Item 1"
        itemPrice="$9.99"
        itemImage="image1.jpg"
        isCartScreen={true}
        onIncQty={onIncQtyMock}
        onDecQty={onDecQtyMock}
        value={2}
        onCart={onCartMock}
      />,
    );

    const itemNameElement = getByText('Item 1');
    expect(itemNameElement).toBeDefined();

    const itemPriceElement = getByText('$9.99');
    expect(itemPriceElement).toBeDefined();

    const stepperValueElement = getByText('2');
    expect(stepperValueElement).toBeDefined();
  });

  it('renders Remove button and calls onDeleteItem handler on cart screen', () => {
    const onDeleteItemMock = jest.fn();
    const onCartMock = jest.fn();
    const {getByText, getByTestId} = render(
      <CardView
        itemName="Item 1"
        itemPrice="$9.99"
        itemImage="image1.jpg"
        isCartScreen={true}
        onDeleteItem={onDeleteItemMock}
        onCart={onCartMock}
      />,
    );

    const removeButton = getByTestId('delete-button');
    expect(removeButton).toBeDefined();
    fireEvent.press(removeButton);
    expect(onDeleteItemMock).toHaveBeenCalled();
  });
  it('renders correctly', () => {
    const tree = render(<CardView {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
