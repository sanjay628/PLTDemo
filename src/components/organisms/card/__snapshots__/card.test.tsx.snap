// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardView renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "white",
      "borderRadius": 24,
      "elevation": 10,
      "flexDirection": "row",
      "height": 300,
      "marginVertical": 16.428571428571427,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 8,
        "width": 8,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 10,
      "width": "90%",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "backgroundColor": "#EBEBEB",
        "borderRadius": 20,
        "height": 240,
        "marginLeft": 15,
        "width": 240,
      }
    }
  />
  <View
    style={
      Object {
        "height": "100%",
        "width": "60%",
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontSize": 28,
            "fontWeight": "600",
          },
          Object {
            "marginLeft": 15,
            "marginTop": 32.857142857142854,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    />
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 28,
            "fontWeight": "600",
          },
          Object {
            "marginLeft": 15,
            "marginTop": 32.857142857142854,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#9191A1",
          "borderRadius": 40,
          "bottom": 16.428571428571427,
          "elevation": 5,
          "height": 82,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 15,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 8,
            "width": 8,
          },
          "shadowOpacity": 0.6,
          "shadowRadius": 10,
          "width": 200,
        }
      }
      testID="submit-button"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "600",
            },
            undefined,
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        Add to Cart
      </Text>
    </View>
  </View>
</View>
`;
